isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -278,18 +281,52 @@ namespace isobus
 		}
 	}
 
-	void VirtualTerminalClient::register_auxiliary_input_event_callback(AuxiliaryInputCallback value)
+	void VirtualTerminalClient::register_auxiliary_function_event_callback(AuxiliaryFunctionCallback value)
 	{
-		auxiliaryInputCallbacks.push_back(value);
+		auxiliaryFunctionCallbacks.push_back(value);
 	}
 
-	void VirtualTerminalClient::remove_auxiliary_input_event_callback(AuxiliaryInputCallback value)
+	void VirtualTerminalClient::remove_auxiliary_function_event_callback(AuxiliaryFunctionCallback value)
 	{
-		auto callbackLocation = std::find(auxiliaryInputCallbacks.begin(), auxiliaryInputCallbacks.end(), value);
+		auto callbackLocation = std::find(auxiliaryFunctionCallbacks.begin(), auxiliaryFunctionCallbacks.end(), value);
 
-		if (auxiliaryInputCallbacks.end() != callbackLocation)
+		if (auxiliaryFunctionCallbacks.end() != callbackLocation)
 		{
-			auxiliaryInputCallbacks.erase(callbackLocation);
+			auxiliaryFunctionCallbacks.erase(callbackLocation);
+		}
+	}
+
+	void VirtualTerminalClient::set_auxiliary_input_model_identification_code(std::uint16_t modelIdentificationCode)
+	{
+		if (state == StateMachineState::Disconnected)
+		{
+			ourModelIdentificationCode = modelIdentificationCode;
+		}
+		else
+		{
+			CANStackLogger::CAN_stack_log("[AUX-N] Error setting model identification code... can only be changed when disconnected");
+		}
+	}
+
+	void VirtualTerminalClient::update_auxiliary_input(const std::uint16_t auxiliaryInputID, const std::uint16_t value1, const std::uint16_t value2, const bool controlLocked)
+	{
+		/// @todo remove this temporary workaround when the VT client can read which objects are passed to the VT (#65)
```

Comment:
We should do the adding of new inputs to `ourAuxiliaryInputs` when parsing/uploading the object pool

Reply:
Instead of parsing them from the object_pool (probably resource consuming and complicated task), you have to register the input object ids manually with setters.